Reflavor update

Mattkittan opened the pull request:
Allows the Reflavor command to apply to class and background.

Mattkittan committed:
Reflavor update

Allows the Reflavor command to apply to class and background.

MorizSchildorfer commented on cogs/char.py:
```diff
@@ -3178,15 +3196,30 @@ async def image(self,ctx, char, url):
             return
-
-    
+        rtype = rtype.lower()      
+        if(rtype != "race" and rtype != "background" and rtype != "class"): #makes sure that the user put in a valid reflavor type, in a messy way that doesn't fit the format of our other error messages.
+            await ctx.channel.send(content=f'***{rtype}*** is an invalid reflavor type.')
+            return
+            
         channel = ctx.channel
```
Instead of limiting it to only 3 possible inputs make 3 command linking to a new function. That will also allow us to easily keep the formatting standard for keys in the DB (first letter capitalized)

Mattkittan replied:
Would you suggest the command names be $race $class and $background respectively?

MorizSchildorfer replied:
$reflavor race etc...
make $reflavor a group

MorizSchildorfer commented on cogs/char.py:
```diff
@@ -3178,15 +3196,30 @@ async def image(self,ctx, char, url):
-        if( len(new_race) > 20 or len(new_race) <1):
+    async def reflavor(self,ctx, rtype, char, *, new_flavor):
+        if( len(new_flavor) > 20 or len(new_flavor) <1):
             await ctx.channel.send(content=f'The new race must be between 1 and 20 symbols.')
```
you missed this line

MorizSchildorfer reviewed (changes requested):
Where is your function for updating the DB?

Mattkittan commented:
> Where is your function for updating the DB?

updatedatabase

MorizSchildorfer commented:
> > Where is your function for updating the DB?
> 
> updatedatabase

I do not see it within the PR

Mattkittan commented:
> > > Where is your function for updating the DB?
> > 
> > 
> > updatedatabase
> 
> I do not see it within the PR

Line 3203

Mattkittan committed:
Update to Reflavor PR

Capitalized the keys, and split the Reflavor command into 3 subcommands, which each call reflavorKernel.

Mattkittan committed:
Update char.py

MorizSchildorfer commented on cogs/admin.py:
```diff
@@ -205,7 +205,25 @@ async def reflavorList(self, ctx):
+        backgroundList = []
+        i = 0
+        while i < len(rfarray):
+            if 'race' in rfarray[i]:
+                raceList.append(rfarray[i]['Race'])
+            if 'class' in rfarray[i]:
+                classList.append(rfarray[i]['Class'])
+            if 'background' in rfarray[i]:
+                backgroundList.append(rfarray[i]['Background'])
```
update the if statements

MorizSchildorfer commented on cogs/char.py:
```diff
@@ -3180,35 +3203,65 @@ async def image(self,ctx, char, url):
 
-    
+        for line in db.players.find({ "Reflavor": { "$exists": 'true' } }):
+            db.players.update_one({"Reflavor": line["Reflavor"]}, [{"$set": {"Reflavor": {"Race": line["Reflavor"]}}}])
```
Add a type check in order to safeguard this command

Mattkittan committed:
More fixes
